src/core/algorithms/fd/fdep/fd_tree_element.cpp

```diff
@@ -261,15 +262,15 @@ void FDTreeElement::TransformTreeFdCollection(std::bitset<kMaxAttrNum>& active_p
             }
             Vertical lhs(&scheme, lhs_bitset);
             Column rhs(&scheme, scheme.GetColumn(attr - 1)->GetName(), attr - 1);
-            fd_collection.emplace_back(FD{lhs, rhs});
+            if (lhs.GetArity() <= max_lhs) fd_collection.emplace_back(FD{lhs, rhs});
```

Comment:
`lhs.GetArity()` is the same for all iterations of this loop and equals `active_path.count()`. There is no need to execute this loop and descend to children if `active_path.count() > max_lhs`. In such case we can return right away in the beginning of the function and by far shorten the traversal